pkg/auth/serviceaccount.go

```diff
@@ -94,3 +97,30 @@ func EnsureOIDCServiceAccountExistsForResource(ctx context.Context, serviceAccou
 
 	return nil
 }
+
+type OIDCStatusMarker interface {
+	MarkOIDCIdentityCreatedSucceeded()
+	MarkOIDCIdentityCreatedSucceededWithReason(reason, messageFormat string, messageA ...interface{})
+	MarkOIDCIdentityCreatedFailed(reason, messageFormat string, messageA ...interface{})
+	MarkStatus(authStatus *duckv1.AuthStatus)
```

Comment:
Instead of requiring all the resources to provide a `MarkStatus` method, we could also pass a callback to `OIDCAuthStatusUtility` to set the status e.g. via:
```
func OIDCAuthStatusUtility(... setAuthStatus func(a *duckv1.AuthStatus)) {
   ...
   setAuthStatus(&duckv1.AuthStatus{
       ServiceAccountName: &saName,
    })
    ...
```

and then in the reconcilers for example:
```
auth.OIDCAuthStatusUtility(ctx, r.serviceAccountLister, r.kubeClientSet, v1.SchemeGroupVersion.WithKind("ApiServerSource"), source.ObjectMeta, &source.Status, func(as *duckv1.AuthStatus) {
    source.Status.AuthStatus = as
});
```